account_move_cutoff/models/account_move_line.py

```diff
@@ -0,0 +1,315 @@
+        self.ensure_one()
+        amounts = dict()
+        line_periods = self._generate_monthly_periods(self.start_date, self.end_date)
+        for period in line_periods:
+            amounts[period] = abs(self.balance) / len(line_periods)
+        return amounts
+
+    def _get_amounts_per_periods_monthly_prorata_temporis(self):
+        self.ensure_one()
+        amounts = dict()
+        line_periods = self._generate_monthly_periods(self.start_date, self.end_date)
+        line_periods_factors = dict.fromkeys(line_periods, 1)
+        last_day_first_period = self._last_day_of_month(line_periods[0])
+        # use of +1 because service days includes start and end dates [start_date, end_date]
+        line_periods_factors[line_periods[0]] = (
+            (last_day_first_period - self.start_date).days + 1
+        ) / last_day_first_period.day
+        line_periods_factors[line_periods[-1]] = (
+            self.end_date.day / self._last_day_of_month(line_periods[-1]).day
+        )
+
+        sum_factor = sum([factor for _period, factor in line_periods_factors.items()])
+        for period, factor in line_periods_factors.items():
+            amounts[period] = abs(self.balance) * factor / sum_factor
+
+        return amounts
+
+    def _get_deferred_amounts_by_period(self, periods):
+        amounts_per_line_and_periods = []
+        for line in self:
+            amounts_per_line_and_periods.append(
+                (line, line._get_amounts_per_periods(periods))
+            )
+        return amounts_per_line_and_periods
+
+    def _get_period_start_end_dates(self, period):
+        last_period_day = self._last_day_of_month(period)
+        if self.start_date > last_period_day or self.end_date < period:
+            start_date = None
+            end_date = None
+        else:
+            start_date = max(self.start_date, period)
+            end_date = min(self.end_date, self._last_day_of_month(period))
+        return start_date, end_date
+
+    def _get_deferred_expense_revenue_account_move_line_labels(self, is_cutoff=None):
+        if is_cutoff:
+            return _("Deferred incomes of %s (%s): %s") % (
+                self.move_id.name,
+                self.date.strftime("%m %Y"),
+                self.name,
+            )
+        else:
+            return _("Adjust deferred incomes of %s (%s): %s") % (
+                self.move_id.name,
+                self.date.strftime("%m %Y"),
+                self.name,
+            )
+
+    def _prepare_entry_lines(self, new_move, period, amount, is_cutoff=True):
+        self.ensure_one()
+        if amount == 0:
+            return self.env["account.move.line"].browse()
+        # deferred_line = self.copy()
+        # deferred_line.account_id = self.deferred_accrual_account_id
+        # deferred_line.tax_ids = False
+        # revenue_or_expense_line = self.copy()
+        reported_credit = reported_debit = 0
+        if self.credit > 0:
```

Comment:
should we use float_compare with proper account decimal precision there?

Reply:
done by https://github.com/OCA/account-closing/pull/256/commits/8466aea6a0701db797bf212739de6776e273b394